src/store/reducers/movies.js

```diff
@@ -1,23 +1,45 @@
-import { GET_MOVIES_PENDING, GET_MOVIES_SUCCESS } from '../actionTypes';
+import { normalizeData } from '../../utils/index';
+import {
+  GET_MOVIES_PENDING, GET_MOVIES_SUCCESS, GET_MOVIE_SUCCESS, GET_MOVIE_PENDING,
+} from '../actionTypes';
 
 const defaultState = {
-  data: [],
+  byId: {},
   page: 1,
   totalPages: 1,
+  isLoading: false,
 };
 
 export const movies = (state = defaultState, { type, payload }) => {
   switch (type) {
     case GET_MOVIES_PENDING:
-      return { ...state };
+      return { ...state, isLoading: true };
     case GET_MOVIES_SUCCESS:
-      return { ...state, data: payload.movies, totalPages: payload.totalPages };
+      return {
+        ...state,
+        totalPages: payload.totalPages,
+        byId: normalizeData(payload.movies),
+        isLoading: false,
+      };
+    case GET_MOVIE_PENDING:
+      return { ...state, isLoading: true };
+    case GET_MOVIE_SUCCESS:
```

Comment:
also add `GET_MOVIES_ERROR` and `GET_MOVIE_ERROR` actions where you will stop loading